job_bundles/README.md

```diff
@@ -0,0 +1,315 @@
+of your Job. The files are described in the following subsections.
+
+### Elements - Job Template
+
+The Job Template file defines the runtime environment and the processes that will run
+as part of an AWS Deadline Cloud Job. It can be parameterized so that the same template can be used to
+create Jobs that differ only in their input values; much like a function or template in your favourite programming
+langauge. 
+
+
+> **NOTE:** You can also setup [Open Job Description external environments](https://github.com/OpenJobDescription/openjd-specifications/wiki/2023-09-Template-Schemas#12-environment-template)
+to Amazon Deadline Cloud Queues (see [Creating a queue environment](https://docs.aws.amazon.com/deadline-cloud/latest/userguide/create-queue-environment.html)
+in the Amazon Deadline Cloud user guide). Queue environments combine with the job template to define the job processes.
+
+
+For example, the [Job Template for the `blender_render` sample](https://github.com/aws-deadline/deadline-cloud-samples/blob/mainline/job_bundles/blender_render/template.yaml)
+defines input parameters like `BlenderSceneFile` which is a file path:
+
+```yaml
+- name: BlenderSceneFile
+  type: PATH
+  objectType: FILE
+  dataFlow: IN
+  userInterface:
+    control: CHOOSE_INPUT_FILE
+    label: Blender Scene File
+    groupLabel: Render Parameters
+    fileFilters:
+    - label: Blender Scene Files
+      patterns: ["*.blend"]
+    - label: All Files
+      patterns: ["*"]
+  description: >
+    Choose the Blender scene file you want to render. Use the 'Job Attachments' tab
+    to add textures and other files that the job needs.
+```
+
+`userInterface`, `objectType`, and `dataFlow` are optional Job Parameter properties that the Deadline Cloud CLI 
+understands when present in a Job Template within a Job Bundle. 
+
+`userInterface` properties control the behaviour and appearance of fields in automatically generated Job submission UIs; both via 
+the `deadline bundle gui-submit` command line, and within Deadline Cloud submitters for applications
+such as the [Autodesk Maya submitter](https://github.com/aws-deadline/deadline-cloud-for-maya). 
+In this example, the UI widget for inputting a value for `BlenderSceneFile` will be a file-selection dialog that
+allows filtering to see only Blender's `.blend` files or all files, and within a widget group called "Render Parameters":
+
+![BlenderSceneFile UI Widget](../.images/blender_submit_scene_file_widget.png)
+
+See the [gui_control_showcase sample](https://github.com/aws-deadline/deadline-cloud-samples/tree/mainline/job_bundles/gui_control_showcase)
+for additional example uses of the `userInterface` property.
+
+The `objectType` and `dataFlow` properties control the behaviour of Deadline Cloud's 
+[Job Attachments feature](https://docs.aws.amazon.com/deadline-cloud/latest/userguide/storage-job-attachments.html) when submitting
```

Comment:
While this is true, could we start by describing the semantics the properties have, and then explain that job attachments uses those semantics? We want other systems to use those semantics as well, and those other systems aren't necessarily job attachments.